pkg/cluster/namespace.go

```diff
@@ -31,18 +32,46 @@ const (
 
 type namespaceService struct {
 	client ctrlpb.NamespaceControllerClient
+	cache  sync.Map
 }
 
 func (ns *namespaceService) GetNamespace(ctx context.Context, id uint64) (*metapb.Namespace, error) {
-	return ns.client.GetNamespace(ctx, &ctrlpb.GetNamespaceRequest{Id: id})
+	v, exist := ns.cache.Load(id)
+	if exist {
+		return v.(*metapb.Namespace), nil
+	}
+	pb, err := ns.client.GetNamespace(ctx, &ctrlpb.GetNamespaceRequest{Id: id})
+	if err != nil {
+		return nil, err
+	}
+	ns.cache.Store(id, pb)
```

Comment:
when the namespace is deleted the cache need to be cleaned, so if uses id it can't cache?